src/note.rs

```diff
@@ -0,0 +1,394 @@
+use kelvin::{ByteHash, Content, Sink, Source};
+use std::convert::TryFrom;
+use std::convert::TryInto;
+use std::io::{self, Read, Write};
+
+use dusk_pki::jubjub_decode;
+use dusk_pki::Ownable;
+use dusk_pki::{PublicSpendKey, SecretSpendKey, StealthAddress, ViewKey};
+
+use dusk_plonk::jubjub::{dhke, GENERATOR_EXTENDED, GENERATOR_NUMS_EXTENDED};
+use poseidon252::cipher::PoseidonCipher;
+use poseidon252::sponge::sponge::sponge_hash;
+use rand::Rng;
+
+use crate::{
+    chunk_of, BlsScalar, Error, JubJubAffine, JubJubExtended, JubJubScalar,
+};
+
+const ENCRYPTED_DATA_SIZE: usize = 96;
```

Comment:
If that's refering to the `PoseidonCipher` I think the constant should be defined in `Poseidon252` repo. Not here.